### PR TITLE
fix(indexer): tolerate reverted balance refresh

### PR DESCRIPTION
**Pull request by fewensa**
## Summary
- skip a reverted `balanceOf` write during onchain refresh when the same task has a successful power read
- keep generic chain tool behavior strict; only refresh value assembly/apply treats this as a partial success
- use effective refresh flags so skipped balance reads do not overwrite contributor balances or delegate relation power
- add an Ark-style regression test where aggregate multicall is out of gas, direct `balanceOf` reverts, and `getVotes` succeeds with zero

## Tests
- `cargo fmt --manifest-path /code/helixbox/degov/Cargo.toml --check`
- `cargo test --manifest-path /code/helixbox/degov/Cargo.toml -p degov-datalens-indexer test_onchain_refresh_skips_reverted_balance_when_power_read_succeeds -- --nocapture`
- `cargo test --manifest-path /code/helixbox/degov/Cargo.toml -p degov-datalens-indexer onchain_refresh --lib`
- `cargo test --manifest-path /code/helixbox/degov/Cargo.toml -p degov-datalens-indexer evm_rpc_chain_tool --lib`

### CODE REVIEW

File: apps/indexer/src/onchain/refresh.rs [hunk 1]
```diff
@@ -1609,8 +1609,12 @@ where
                     Some(value) => Some(value),
                     None => {
                         if let Some(messages) = failures_by_key.get(&key) {
-                            task_failures.extend(messages.iter().cloned());
-                            None
+                            if power.is_some() && is_reverted_balance_read_failure(messages) {
+                                Some("0".to_owned())
```
[the remark this answers is not in the document]
Comment (fewensa) [reply apps/indexer/src/onchain/refresh.rs]: Fixed in the latest push. The refresh now treats this as a partial success instead of zeroing the balance: a reverted balance read is skipped when a power read succeeds, effective refresh flags prevent balance/delegate-relation writes without an actual balance value, and the regression test now asserts `balance == None` rather than `0`.
